src/topology/vector_bundle.lean

```diff
@@ -503,3 +536,279 @@ lemma to_topological_vector_bundle :
 end topological_vector_prebundle
 
 end
+
+/-! ### Direct sum of two vector bundles over the same base -/
+
+namespace topological_vector_bundle
+
+/-- The direct sum of the topological vector bundles `E₁` and `E₂`.  Type synonym for
+`λ x, E₁ x × E₂ x`. -/
+@[derive [topological_space, add_comm_monoid, module R], nolint unused_arguments]
+def vector_bundle_prod {B : Type*}
+  (F₁ : Type*) (E₁ : B → Type*) [Π x, add_comm_monoid (E₁ x)] [Π x, module R (E₁ x)]
+  [Π x : B, topological_space (E₁ x)]
+  (F₂ : Type*) (E₂ : B → Type*) [Π x, add_comm_monoid (E₂ x)] [∀ x, module R (E₂ x)]
+  [Π x : B, topological_space (E₂ x)]
+  (x : B) :=
+E₁ x × E₂ x
+
+instance {B : Type*}
+  (F₁ : Type*) (E₁ : B → Type*) [Π x, add_comm_monoid (E₁ x)] [Π x, module R (E₁ x)]
+  [Π x : B, topological_space (E₁ x)]
+  (F₂ : Type*) (E₂ : B → Type*) [Π x, add_comm_monoid (E₂ x)] [∀ x, module R (E₂ x)]
+  [Π x : B, topological_space (E₂ x)]
+  (x : B) :
+  inhabited (vector_bundle_prod R F₁ E₁ F₂ E₂ x) :=
+⟨0⟩
+
+variables (F₁ : Type*) [topological_space F₁] [add_comm_monoid F₁] [module R F₁]
+  (E₁ : B → Type*) [Π x, add_comm_monoid (E₁ x)] [Π x, module R (E₁ x)]
+  [Π x : B, topological_space (E₁ x)] [topological_space (total_space E₁)]
+
+variables (F₂ : Type*) [topological_space F₂] [add_comm_monoid F₂] [module R F₂]
+  (E₂ : B → Type*) [Π x, add_comm_monoid (E₂ x)] [Π x, module R (E₂ x)]
+  [Π x : B, topological_space (E₂ x)] [topological_space (total_space E₂)]
+
+namespace pretrivialization
+variables (e₁ : trivialization R F₁ E₁) (e₂ : trivialization R F₂ E₂)
+include e₁ e₂
+variables {R F₁ E₁ F₂ E₂}
+
+/-- Given trivializations `e₁`, `e₂` for vector bundles `E₁`, `E₂` over a base `B`, the forward
+function for the construction `topological_vector_bundle.pretrivialization.prod`, the induced
+pretrivialization for the direct sum of `E₁` and `E₂`. -/
+def prod.to_fun' : total_space (vector_bundle_prod R F₁ E₁ F₂ E₂) → B × (F₁ × F₂) :=
+λ ⟨x, v₁, v₂⟩, ⟨x, (e₁ ⟨x, v₁⟩).2, (e₂ ⟨x, v₂⟩).2⟩
+
+variables [topological_vector_bundle R F₁ E₁] [topological_vector_bundle R F₂ E₂]
+
+/-- Given trivializations `e₁`, `e₂` for vector bundles `E₁`, `E₂` over a base `B`, the inverse
+function for the construction `topological_vector_bundle.pretrivialization.prod`, the induced
+pretrivialization for the direct sum of `E₁` and `E₂`. -/
+def prod.inv_fun' (p : B × (F₁ × F₂)) : total_space (vector_bundle_prod R F₁ E₁ F₂ E₂) :=
+begin
+  obtain ⟨x, w₁, w₂⟩ := p,
+  refine ⟨x, _, _⟩,
+  { by_cases h : x ∈ e₁.base_set,
+    { exact (e₁.continuous_linear_equiv_at x h).symm w₁ },
+    { exact 0 } },
+  { by_cases h : x ∈ e₂.base_set,
+    { exact (e₂.continuous_linear_equiv_at x h).symm w₂ },
+    { exact 0 } },
```

Comment:
Does something break if you just use `⟨x, (e₁.continuous_linear_equiv_at x h).symm w₁, (e₂.continuous_linear_equiv_at x h).symm w₂⟩`?